driver-core/src/main/com/mongodb/KerberosSubjectProvider.java

```diff
@@ -80,16 +89,32 @@ public KerberosSubjectProvider(@NonNull final String loginContextName) {
     @NonNull
     public synchronized Subject getSubject() throws LoginException {
         if (subject == null || needNewSubject(subject)) {
-            LOGGER.info("Creating new LoginContext and logging in the principal");
-            LoginContext loginContext = new LoginContext(loginContextName);
-            loginContext.login();
-            subject = loginContext.getSubject();
-            LOGGER.info("Login successful");
+            subject = createNewSubject();
         }
         return subject;
     }
 
-    private static boolean needNewSubject(@NonNull final Subject subject) {
+    private Subject createNewSubject() throws LoginException {
+        LoginContext loginContext;
+        try {
+            LOGGER.debug(format("Creating LoginContext with name '%s'", loginContextName));
+            loginContext = new LoginContext(loginContextName);
```

Comment:
Take advantage of the fact that it's the constructor that throws if the name is not found, not the login method